librato/alerts.py

```diff
@@ -0,0 +1,131 @@
+        self.conditions = []
+        for c in conditions:
+            if isinstance(c, Condition):
+                self.conditions.append(c)
+            elif isinstance(c, dict):  # Probably parsing JSON here
+                self.conditions.append(Condition.from_dict(c))
+            else:
+                self.conditions.append(Condition(*c))
+        self.services = []
+        for s in services:
+            if isinstance(s, Service):
+                self.services.append(s)
+            elif isinstance(s, dict):  # Probably parsing JSON here
+                self.services.append(Service.from_dict(s))
+            elif isinstance(s, int):
+                self.services.append(Service(s))
+            else:
+                self.services.append(Service(*s))
+        self.attributes = attributes
+        self.active = active
+        self.rearm_seconds = rearm_seconds
+        self._id = _id
+
+    def add_condition_for(self, metric_name, source='*'):
+        condition = Condition(metric_name, source)
+        self.conditions.append(condition)
+        return condition
+
+    def add_service(self, service_id):
+        self.services.append(Service(service_id))
+
+    @classmethod
+    def from_dict(cls, connection, data):
+        """Returns an alert object from a dictionary item,
+        which is usually from librato's API"""
+        obj = cls(connection,
+                  data['name'],
+                  version=data['version'],
+                  description=data['description'],
+                  conditions=data['conditions'],
+                  services=data['services'],
+                  _id=data['id'],
+                  attributes=data['attributes'])
+        return obj
+
+    def get_payload(self):
+        return {'name': self.name,
+                'attributes': self.attributes,
+                'version': self.version,
+                'description': self.description,
+                'services': [x._id for x in self.services],
+                'conditions': [x.get_payload() for x in self.conditions]}
+    
+    def save(self):
+        self.connection.update_alert(self)
+
+class Condition(object):
+    def __init__(self, metric_name, source='*'):
+        self.metric_name = metric_name
+        self.source = source
+
+    def above(self, threshold, summary_function=None):
+        self.condition_type = 'above'
+        self.summary_function = summary_function
+        self.threshold = threshold
+        self.duration = None
+        return self
+
+    def below(self, threshold, summary_function=None):
+        self.condition_type = 'below'
+        self.summary_function = summary_function
+        self.threshold = threshold
+        self.duration = None
+        return self
+
+    def stops_reporting_for(self, duration, summary_function=None):
+        self.condition_type = 'absent'
+        self.summary_function = summary_function
+        self.duration = duration
+        return self
+
+    def during(self, duration):
```

Comment:
More like https://github.com/mauricioborges/python-librato/blob/adding-alert-support/README.md#alerts where you've added the example `alert.add_condition_for('metric_name').above(1).during(60)`.  The threshold alerts can either be immediate, or have a duration, like "below 42 for 60 seconds".  So I'd prefer `alert.add_condition_for('metric_name').above(1).duration(60)`

Make sense?